substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -1851,6 +1851,18 @@ pub mod pallet {
             farm.farming_policy_limits = limits.clone();
             Farms::<T>::insert(farm_id, farm);
 
+            // Give all the nodes in this farm the policy that is attached
+            for node_id in NodesByFarmID::<T>::get(farm_id) {
+                match Nodes::<T>::get(node_id) {
+                    Some(mut node) => {
+                        let policy = Self::get_farming_policy(&node)?;
+                        node.farming_policy_id = policy.id;
```

Comment:
This should be a bit more refined. Only if the node currently has a "default" policy (i.e. one which is not explicitly linked), should this be changed.